backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala

```diff
@@ -180,13 +180,28 @@ class CHTransformerApi extends TransformerApi with Logging {
   override def getPlanOutput(plan: SparkPlan): Seq[Attribute] = {
     plan match {
       case hash: HashAggregateExec =>
-        CHHashAggregateExecTransformer.getAggregateResultAttributes(
+        val groupingExpressions = hash.output.splitAt(hash.groupingExpressions.size)._1
+        val aggResultAttributes = CHHashAggregateExecTransformer.getAggregateResultAttributes(
           // when grouping expression has alias,
           // output name will be different from grouping expressions,
           // so using output attribute instead of grouping expression
-          hash.output.splitAt(hash.groupingExpressions.size)._1,
+          groupingExpressions,
           hash.aggregateExpressions
         )
+        if (aggResultAttributes.size == hash.output.size) {
```

Comment:
since the output schema has been change. please to consider following case
```sql
select a, 1 as b, count(1) from t group by a
```

Reply:
done, add a UT